app/controllers/posts_controller.rb

```diff
@@ -48,6 +48,8 @@ def create
                 end || nil
     @post = Post.new(post_params.merge(user: current_user, body: helpers.post_markdown(:post, :body_markdown),
                                        category: @category, post_type: @post_type, parent: @parent))
+    @post.save
```

Comment:
How does this save interact with the subsequent checks that can reject the post (e.g. title too short)?  Is there another step beyond save where the changes get committed?